From 35daaeb956bfc9c0e8c1a858d85a451c5eedcbb3 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 07 Jul 2022 15:05:41 +0300
Subject: [PATCH] [SW-94722] gaudi2: remove clearing msix address reg from ppboot

It is observed that accessing PCIE_DBI registers during pci link
transition can cause a cpu freeze. This issue has been observed
earlier(https://gerrit.habana-labs.com/#/c/101296/). Hence,
clearing msix match address registers after pcie initialization
in preboot.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I5a4367f529b66724de2b0ac6facccf821ec7b37e
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 337f82a..f552a60 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -202,12 +202,6 @@
 	init_arc_mem(0, ARC0_ICCM_FULL_BASE, ARC0_ICCM_SIZE);
 }
 
-static void gaudi2_clear_msix_addr_match(void)
-{
-	regmap_writel(0, mmPCIE_DBI_MSIX_ADDRESS_MATCH_HIGH_OFF);
-	regmap_writel(0, mmPCIE_DBI_MSIX_ADDRESS_MATCH_LOW_OFF);
-}
-
 void hl_plat_early_init(void)
 {
 	/* Functionality used to be done in BTL */
@@ -220,8 +214,6 @@
 		cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN);
 
 	gaudi2_axi_drain_ena();
-
-	gaudi2_clear_msix_addr_match();
 
 	/* set reset source */
 	hl_reset_src_init();
